packages/js/product-editor/src/components/custom-fields/custom-field-name-control/custom-field-name-control.tsx

```diff
@@ -0,0 +1,249 @@
+	function ForwardedCustomFieldNameControl(
+		{
+			allowReset,
+			className,
+			help,
+			hideLabelFromVision,
+			label,
+			messages,
+			value,
+			onChange,
+			onBlur,
+			...props
+		}: CustomFieldNameControlProps,
+		ref: ForwardedRef< HTMLInputElement >
+	) {
+		const comboboxRef = useRef< HTMLInputElement >();
+		const id = useInstanceId(
+			CustomFieldNameControl,
+			'woocommerce-custom-field-name'
+		);
+
+		const [ customFieldNames, setCustomFieldNames ] = useState<
+			ComboboxControl.Props[ 'options' ]
+		>( [] );
+
+		useLayoutEffect(
+			/**
+			 * The Combobox component does not expose the ref to the
+			 * internal native input element removing the ability to
+			 * focus the element when validating it in the context
+			 * of a form
+			 */
+			function initializeRefs() {
+				comboboxRef.current = document.querySelector(
+					`.${ id } [role="combobox"]`
+				) as HTMLInputElement;
+
+				if ( ref ) {
+					if ( typeof ref === 'function' ) {
+						ref( comboboxRef.current );
+					} else {
+						ref.current = comboboxRef.current;
+					}
+				}
+			},
+			[ id, ref ]
+		);
+
+		const { attrs, events } = useMemo(
+			function splitAttrsAndEvents() {
+				return Object.entries( props ).reduce<
+					Record< string, Record< string, unknown > >
+				>(
+					( current, [ propName, propValue ] ) => {
+						if ( propValue !== undefined ) {
+							if ( propName.startsWith( 'on' ) ) {
+								const eventName = propName
+									.substring( 2 )
+									.toLowerCase();
+								current.events[ eventName ] =
+									propValue as never;
+							} else {
+								current.attrs[ propName ] = propValue as never;
+							}
+						}
+
+						return current;
+					},
+					{ attrs: {}, events: {} }
+				);
+			},
+			[ props ]
+		);
+
+		useEffect(
+			/**
+			 * The Combobox component does not expose any attribute
+			 * of the internal native input element removing the ability
+			 * to set attribute's values like name important in the
+			 * context of form submission
+			 */
+			function initializeAttrs() {
+				Object.entries( attrs ).forEach(
+					( [ propName, propValue ] ) => {
+						comboboxRef.current?.setAttribute(
+							propName,
+							`${ propValue }`
+						);
+					}
+				);
+			},
+			[ attrs ]
+		);
+
+		useEffect(
+			/**
+			 * The Combobox component does not expose any event
+			 * of the internal native input element removing the ability
+			 * to focus the element and other things related also to
+			 * validations
+			 */
+			function initializeEvents() {
+				Object.entries( events ).forEach(
+					( [ propName, propValue ] ) => {
+						comboboxRef.current?.addEventListener(
+							propName,
+							propValue as () => void
+						);
+					}
+				);
+
+				return () => {
+					Object.entries( events ).forEach(
+						( [ propName, propValue ] ) => {
+							comboboxRef.current?.removeEventListener(
+								propName,
+								propValue as () => void
+							);
+						}
+					);
+				};
```

Comment:
When I comment out this and the initialize attribute functions I don't notice any difference in how the dropdown behaves, are the above really necessary?
Do you have an exact example of what it provides?